app/src/main/java/com/example/rwothoromo/developers/presenter/GithubPresenter.java

```diff
@@ -0,0 +1,54 @@
+package com.example.rwothoromo.developers.presenter;
+
+import com.example.rwothoromo.developers.model.Data;
+import com.example.rwothoromo.developers.model.GithubUser;
+import com.example.rwothoromo.developers.service.GithubService;
+import com.example.rwothoromo.developers.view.GithubUserView;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * This class is a service that leverages the GithubApi interface.
+ * It represents a view of the data retrieved.
+ */
+public class GithubPresenter {
+
+	private GithubUserView githubUserView;
+	private GithubService githubService;
+
+	public GithubPresenter(GithubUserView view) {
+		this.githubUserView = view;
+
+		if (this.githubService == null) {
+			this.githubService = new GithubService();
+		}
+	}
+
+	public void getGithubUsers() {
+		githubService.getAPI().getResults()
+				.enqueue(new Callback<Data>() {
+					@Override
+					public void onResponse(Call<Data> call, Response<Data> response) {
+						Data data = response.body();
+
+						if (data != null && data.getGithubUsersResponse() != null) {
+							List<GithubUser> result = data.getGithubUsersResponse().getResult();
+							githubUserView.githubUsersReady(result);
+						}
+					}
+
+					@Override
+					public void onFailure(Call<Data> call, Throwable t) {
+						try {
+							throw new InterruptedException("Failed to fetch GitHub users!");
```

Comment:
@Rwothoromo, you could have the view display an error message here instead of throwing an error you are catching in the same block (this looks kind of useless to me).

If you need to know the cause of the error for debugging purposes, just print the stack trace through the `Throwable` parameter.

I guess the implementation of the `View` interface would come much later?

Reply:
@idrisadetunmbi I am a little confused here concerning the error.

Also, the `View` comes in later but will instead be a `GithubAdapter` instead of `GithubUserView`.

Reply:
The resolution was:
```
@Override
public void onFailure(Call<Data> call, Throwable t) {

	githubUserView.failedDataRetrieval();

}
```